src/ensembl/src/content/app/browser/track-panel/track-panel-modal/modal-views/TrackPanelBookmarks.tsx

```diff
@@ -3,50 +3,62 @@ import { Link } from 'react-router-dom';
 import { connect } from 'react-redux';
 
 import { RootState } from 'src/store';
-import { getExampleEnsObjects } from 'src/ens-object/ensObjectSelectors';
-import { fetchExampleEnsObjectsData } from 'src/ens-object/ensObjectActions';
+import { EnsObject } from 'src/ens-object/ensObjectTypes';
+import { GenomeInfo } from 'src/genome/genomeTypes';
+
+import { getBrowserActiveGenomeId } from '../../../browserSelectors';
+import {
+  getGenomeExampleEnsObjects,
+  getGenomeInfo
+} from 'src/genome/genomeSelectors';
+import { fetchGenomeExampleEnsObjects } from 'src/genome/genomeActions';
+import * as urlFor from 'src/shared/helpers/urlHelper';
 
 import styles from '../TrackPanelModal.scss';
 
 type StateProps = {
-  exampleEnsObjects: any;
+  activeGenomeId: string;
+  genomeInfo: GenomeInfo;
+  genomeExampleEnsObjects: EnsObject[];
 };
 
 type DispatchProps = {
-  fetchExampleEnsObjectsData: () => void;
+  fetchGenomeExampleEnsObjects: (genomeId: string) => void;
 };
 
 type OwnProps = {};
 
 type TrackPanelBookmarksProps = StateProps & DispatchProps & OwnProps;
 
 export const TrackPanelBookmarks = (props: TrackPanelBookmarksProps) => {
-  const exampleObjectsTotal = Object.keys(props.exampleEnsObjects).length;
-
   useEffect(() => {
-    if (exampleObjectsTotal === 0) {
-      props.fetchExampleEnsObjectsData();
+    const {
+      activeGenomeId,
+      genomeExampleEnsObjects,
+      fetchGenomeExampleEnsObjects
+    } = props;
+
+    if (
+      genomeExampleEnsObjects.length === 0 ||
+      genomeExampleEnsObjects[0].genome_id !== activeGenomeId
+    ) {
+      fetchGenomeExampleEnsObjects(activeGenomeId);
     }
-  }, [props.exampleEnsObjects]);
+  }, []);
```

Comment:
Not sure what's going on here:
- should the possible data fetching happen only once, when the component is mounted? Are there any props that, when changed, should prompt re-fetching?
- how can it happen that `genomeExampleEnsObjects[0].genome_id !== activeGenomeId`? Shouldn't example objects always correspond to the active genome id?

Reply:
The data should be refetched if the focused object changes. This can happen if a user clicks on an example object link in the bookmarks modal of the track panel.

As for the check, it is to refetch data when the genome ID changes. In hindsight, this probably should be in a `useEffect` that fires whenever the genome ID changes.